fretboard/src/main/java/mozilla/components/service/fretboard/SyncConfig.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fretboard
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Represents the configuration for
+ * scheduling experiment downloads
+ */
+data class SyncConfig(
+    val immediately: Boolean = true,
+    val minimumLatency: Long? = null,
+    val overrideDeadline: Long? = null,
+    val persisted: Boolean = false,
+    val periodicity: Long? = TimeUnit.DAYS.toMillis(1),
+    val backoff: Backoff? = null,
+    val requiresUnmeteredNetwork: Boolean = false,
+    val requiresCharging: Boolean = false,
+    val requiresDeviceIdle: Boolean = false
+) {
```

Comment:
I wonder if it's worth having this abstraction of a sync configuration. This is very jobscheduler-like and something that may not make sense with a different system. Does a consumer really need to interact with the scheduler via a shared interface?

At the app level we already know the scheduler implementation we use (e.g. job scheduler). Maybe I could just interact with the scheduler directly?

```Kotlin
val scheduler = ....
scheduler.scheduleUpdate(....)
```

With that it's up to the scheduler to expose whatever services it offers.